kernel/api/src/main/java/org/sakaiproject/tasks/api/TaskAssigned.java

```diff
@@ -0,0 +1,48 @@
+package org.sakaiproject.tasks.api;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+
+import org.sakaiproject.springframework.data.PersistableEntity;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@Entity
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+@Table(name = "TASKS_ASSIGNED", indexes = {
+    @Index(name = "IDX_TASKS_ASSIGNED", columnList = "TASK_ID")
+})
+public class TaskAssigned implements PersistableEntity<Long> {
+
+	public static final String USER = "user";
+	public static final String SITE = "site";
+	public static final String GROUP = "group";
+
+	@Id
+	@Column(name = "ID")
+	@GeneratedValue(strategy = GenerationType.AUTO, generator = "tasks_assigned_id_sequence")
+	@SequenceGenerator(name = "tasks_assigned_id_sequence", sequenceName = "TASKS_ASSIGNED_S")
+	@EqualsAndHashCode.Include
+	private Long id;
+
+	@ManyToOne
+	@JoinColumn(name = "TASK_ID", nullable = false)
+	private Task task;
+
+	@Column(name = "TYPE", length = 5, nullable = false)
+	private String type;
```

Comment:
You should make this an enum. Have a look at the TopicVisibility and Topic classes in the conversations project.

Reply:
Also, TYPE is a reserved word in Mysql - you should probably use something like ASSIGNED_TYPE

Reply:
As Adrian said use enum and " @Enumerated(EnumType.STRING)".

public enum AssignationType {user, site, group}

	@Column(name = "ASSIGNATION_TYPE", length = 5, nullable = false)
        @Enumerated(EnumType.STRING)
	private AssignationType type;

Reply:
Or even just use an ordinal to use less space in the db. That doesn't seem like much, but when you have massive database hosted on Amazon it can mount up.